src/Illuminate/Validation/Validator.php

```diff
@@ -617,6 +617,18 @@ protected function validateRequired($attribute, $value)
     }
 
     /**
+     * Validate that an attribute exists even if not filled.
+     *
+     * @param  string  $attribute
+     * @param  mixed   $value
+     * @return bool
+     */
+    protected function validatePresent($attribute, $value)
+    {
+        return Arr::get($this->data, $attribute, '__NO_KEY__') != '__NO_KEY__';
```

Comment:
You are right, I used `Arr::has` to simplify it in https://github.com/laravel/framework/pull/12489 for both rules (present & filled).